cmd/oras/internal/display/status/progress/speed.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package progress
+
+type speedWindow struct {
+	point []int64
```

Comment:
According to your algorithm for accuracy, a point should be a pair of `(time, offset)`. Then the average time is $$\text{Average Time} = \frac{\text{head}.offset - \text{tail}.offset}{\text{head}.time - \text{tail}.time}$$.

Reply:
We can discuss the algorithm offline if needed

Reply:
yes you are right, will change accordingly.